src/mechanism.cpp

```diff
@@ -170,13 +170,31 @@ size_t zmq::mechanism_t::add_basic_properties (unsigned char *buf,
                         options.routing_id, options.routing_id_size);
     }
 
+
+#ifdef ZMQ_BUILD_DRAFT_API
```

Comment:
There's no need to wrap the implementation in ifdefs, it's enough that the symbol is not available - if one hacks it through, it's their problem if it changes

So you can remove all the ifdefs, but leave the test in the DRAFT section of the makefile (and also please add it to tests/CMakeList)

Reply:
Ah I see, will fix. I thought it was correctly added to tests/CMakeList, did I miss adding it somewhere else in there? 
https://github.com/eponsko/libzmq/blob/master/tests/CMakeLists.txt#L138